src/pages/workspaces/workspace/notebooks/modals/CloudEnvironmentModal.js

```diff
@@ -0,0 +1,404 @@
+        })
+      default:
+        return defaultIcon(toolLabel)
+    }
+  }
+
+  // TODO: multiple runtime: build component around this logic for a multiple runtime approach. see getCostForTool for example usage
+  const getRuntimeForTool = toolLabel => Utils.cond([toolLabel === currentRuntimeTool, () => currentRuntime],
+    [Utils.DEFAULT, () => undefined])
+
+  const getToolIcon = toolLabel => Utils.switchCase(toolLabel,
+    [tools.Jupyter.label, () => jupyterLogo],
+    [tools.galaxy.label, () => galaxyLogo],
+    [tools.RStudio.label, () => rstudioLogo])
+
+  // TODO: multiple runtime: this is a good example of how the code should look when multiple runtimes are allowed, over a tool-centric approach
+  const getCostForTool = toolLabel => Utils.cond(
+    [toolLabel === tools.galaxy.label, () => getGalaxyCostTextChildren(currentApp, galaxyDataDisks)],
+    [getRuntimeForTool(toolLabel), () => {
+      const runtime = getRuntimeForTool(toolLabel)
+      const totalCost = runtimeCost(runtime) + _.sum(_.map(persistentDiskCost, persistentDisks))
+      return span([`${runtime.status} (${Utils.formatUSD(totalCost)} / hr)`])
+    }],
+    [Utils.DEFAULT, () => span(['None'])]
+  )
+
+  const isCloudEnvModalDisabled = toolLabel => Utils.cond(
+    [toolLabel === tools.galaxy.label, () => !canCompute || busy || isCurrentGalaxyDiskDetaching(apps) || getIsAppBusy(currentApp)],
+    [Utils.DEFAULT, () => {
+      const runtime = getRuntimeForTool(toolLabel)
+      return runtime ?
+        !canCompute || busy || getIsRuntimeBusy(runtime) :
+        !canCompute || busy || getIsRuntimeBusy(currentRuntime) //TODO: multiple runtimes: change this to not have the last check in the or
+    }]
+  )
+
+  const getToolLaunchClickableProps = toolLabel => {
+    const doesCloudEnvForToolExist = currentRuntimeTool === toolLabel || (currentApp && toolLabel === tools.galaxy.label)
+    // TODO what does cookieReady do? Found it in the galaxy app launch code, is it needed here?
+    const isToolBusy = toolLabel === tools.galaxy.label ? getIsAppBusy(currentApp) || currentApp?.status === 'STOPPED' || currentApp?.status === 'ERROR' : currentRuntime?.status === 'Error'
+    const isDisabled = !doesCloudEnvForToolExist || !cookieReady || !canCompute || busy || isToolBusy || toolLabel === tools.Jupyter.label
+    const baseProps = {
+      'aria-label': `Launch ${toolLabel}`,
+      disabled: isDisabled,
+      style: {
+        ...toolButtonStyles,
+        color: isDisabled ? colors.dark(0.7) : colors.accent()
+      },
+      hover: { backgroundColor: colors.accent(0.2) },
+      tooltip: Utils.cond(
+        [doesCloudEnvForToolExist && !isDisabled, () => 'Launch'],
+        [doesCloudEnvForToolExist && isDisabled && toolLabel !== tools.Jupyter.label, () => 'Launch disabled'],
+        [doesCloudEnvForToolExist && isDisabled && toolLabel === tools.Jupyter.label, () => 'Generic launch disabled for Jupyter. Select a notebook in the analyses tab to launch Jupyter'],
```

Comment:
I eventually understood what the first sentence in the message is trying to convey but the first time I saw it, I had to think what `generic launch` would be. Since it's not something established per se, I'd suggest removing it, and use the space to also mention creating a notebook in case users don't have one already, like so

```suggestion
        [doesCloudEnvForToolExist && isDisabled && toolLabel === tools.Jupyter.label, () => 'Select or create a notebook in the analyses tab to launch Jupyter'],
```